src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,226 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
  */
 ucc_status_t ucc_collective_finalize(ucc_coll_req_h request);
 
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef enum ucc_event_type {
+	UCC_EVENT_COLLECTIVE_POST       = UCC_BIT(0),
+	UCC_EVENT_COLLECTIVE_COMPLETE   = UCC_BIT(1),
+	UCC_EVENT_COMPUTE_COMPLETE      = UCC_BIT(2),
+	UCC_EVENT_OVERFLOW              = UCC_BIT(3)
+} ucc_event_type_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef enum ucc_ee_type {
+    UCC_CUDA_STREAM = UCC_BIT(0),
+	UCC_CPU_THREAD  = UCC_BIT(1)
+} ucc_ee_type_t;
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+typedef struct ucc_event_context {
+    size_t              info_size;
+    void                *info;
+} ucc_event_context_t;
```

Comment:
What is the purpose of event_context_t on its own? Maybe info and info_size should just be part of ucc_event as a substructure?

Reply:
We had more fields in the structure, however, we trimmed it based on the discussion/feedback from the members here. Now, I don't have a strong opinion on whether it should be a separate structure or can be absorbed. Given, it is a style issue, is there a preference? @bureddy

Reply:
looks like moving  them to event struct is cleaner. WDYT?
```
typedef struct ucc_event {
    ucc_event_type_t        ev_type;
    void     *event_context;
    size_t    event_context_size;
    ucc_coll_req_h          req;
} ucc_ev_t;
```